### PR TITLE
adding Time of Flight sensor drivers

### PR DESCRIPTION
**Pull request by chrpro**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
I add the drivers for VL53L0Xupdater Time of Flight sensor

* **What is the current behavior?** (You can also link to an open issue here)
There is no ToF sensor on Truck state

* **What is the new behavior (if this is a feature change)?**
The Truckstate will have access to ToF sensor and store the distance from objects on an integer

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:

### CODE REVIEW

File: sensors/tof/VL53L0Xupdater.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+#ifndef VL53L0XUPDATER_H
+#define VL53L0XUPDATER_H
+
+#include "sensors/tof/tofsensorupdater.h"
+#include "vehicles/truckstate.h"
+
+#include <QTimer>
+
+// connects to AS5600 via i2c bus and polls orientation data (angle) periodically
```
Comment (marvind) [inline sensors/tof/VL53L0Xupdater.h:16]: Seems to me this comment was copy-pasted from another file. Needs update

File: sensors/tof/VL53L0Xupdater.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class VL53L0XUpdater : public ToFSensorUpdater
+{
+public:
+    VL53L0XUpdater(QSharedPointer<VehicleState> vehicleState);
+    //void printSensorInfo();
+    virtual bool setUpdateIntervall(int pollIntervall_ms) override;
+
+private:
+    int mPollIntervall_ms = 500; // interval (ms) to read from AS5600,
```
Comment (marvind) [inline sensors/tof/VL53L0Xupdater.h:25]: Similar to previous comment as this is not for AS5600

File: sensors/tof/VL53L0Xupdater.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+      connect(&mPollTimer, &QTimer::timeout, [this, vehicleState]() {
+         int iDistance = tofReadDistance();
+         if (iDistance < 4096){ // valid range?
+            //qDebug() <<  "Object Distance to trailer = " <<  iDistance << "mm";
+         }
+
+         QSharedPointer<TruckState> truckState = qSharedPointerDynamicCast<TruckState>(vehicleState);
+         if (truckState) {  
+            truckState->setTrailerDistanceToF(iDistance);
```
Comment (marvind) [inline sensors/tof/VL53L0Xupdater.cpp:35]: Please do not report the distance in mm but use SI base units (meters).

File: sensors/tof/tofsensorupdater.h [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/*
+ *     Copyright 2024 RISE Sweden
+ *     Published under GPLv3: https://www.gnu.org/licenses/gpl-3.0.html
+ * 
+ * Abstract interface for receiving tof values e.g., magnetic rotary position sensor
```
Comment (marvind) [inline sensors/tof/tofsensorupdater.h:5]: "magnetic rotary position sensor" is not correct to my understanding

File: vehicles/truckstate.h [hunk 5]
```diff
@@ -54,7 +61,7 @@ class TruckState : public CarState
     double mtrailerwheelbase; // trailer wheelbase
     QSharedPointer<TrailerState> mTrailerState; // trailer created dynamically
     bool mHasTrailer = false;
-
+    int trailer_tof_distance; // distance to nearest object
```
Comment (marvind) [inline vehicles/truckstate.h:64]: Please follow naming conventions and report distance in SI base units (meters).

**Review by marvind — CHANGES_REQUESTED** [review]

**Review by r2avula — APPROVED** [review]
Tested with [waywise_truck.cpp]( https://github.com/RISE-Dependable-Transport-Systems/WayWiseR/blob/6ff50dcc5bad7c9efc8799706c508607a0c4bdfb/waywiser_node/src/waywise_truck.cpp) and works as expected 👍

**Review by marvind — APPROVED** [review]

**Pull request merged**